### ISSUE TITLE
[Q] How to get determinstic result among multiple runs?

### ISSUE THREAD

**Issue by kehuanfeng** [modelscope/ms-swift#2811]
The loss is not consistent among multiple runs  even if we set the seed. 

ms-swift                      3.1.0.dev0

```
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NPROC_PER_NODE=8 \
MAX_PIXELS=1003520 \
swift sft \
    --model /data/models/Qwen/Qwen2-VL-7B-Instruct \
    --train_type full \
    --dataset 'swift/OK-VQA_train#1000' \
    --num_train_epochs 3 \
    --per_device_train_batch_size 1 \
    --learning_rate 1e-4 \
    --gradient_accumulation_steps 4 \
    --logging_steps 5 \
    --seed 2025
```

**Comment by Jintao-Huang** [comment]
`--full_determinism true`

**Comment by kehuanfeng** [comment]
> `--full_determinism true`

Thanks for the comment.  We still see some deviation with above option.
